app/services/report_importer.rb

```diff
@@ -44,6 +44,15 @@ def import
     end
   end
 
+  def scan(save = false)
+    logger.info "Scanning report with: #{report_scanner.join(', ')}"
+    report_scanner.each do |scanner|
```

Comment:
if first scanner finds the origin, e.g. 'Puppet', since we use `each` it continues scanning, since there can be just one origin, I think it would be better to "break" on first match, probably using `any?` instead of `each`

Reply:
:bulb: Now I see where you are getting at. I didn't realize that the importer only processes one report at a time. I'll rewrite it to scan, break when the first `scanner.scan` returns true and and let it be saved afterwards in the `create_report_and_logs`